Mathlib/RingTheory/Polynomial/Wronskian.lean

```diff
@@ -113,4 +114,40 @@ theorem natDegree_wronskian_lt_add {a b : R[X]} (hw : wronskian a b ≠ 0) :
   · exact Polynomial.degree_eq_natDegree ha
   · exact Polynomial.degree_eq_natDegree hb
 
+variable {k : Type*} [Field k]
+
+@[simp]
+theorem dvd_derivative_iff {a : k[X]} : a ∣ derivative a ↔ derivative a = 0 := by
+  constructor
+  · intro h
+    by_cases a_nz : a = 0
+    · rw [a_nz]; simp only [derivative_zero]
+    by_contra deriv_nz
+    have deriv_lt := degree_derivative_lt a_nz
+    have le_deriv := Polynomial.degree_le_of_dvd h deriv_nz
+    have lt_self := le_deriv.trans_lt deriv_lt
+    simp only [lt_self_iff_false] at lt_self
+  intro h; rw [h]; simp
```

Comment:
```suggestion
theorem dvd_derivative_iff {a : k[X]} : a ∣ derivative a ↔ derivative a = 0 where
  mp h := by
    by_cases a_nz : a = 0
    · rw [a_nz]; simp only [derivative_zero]
    by_contra deriv_nz
    have deriv_lt := degree_derivative_lt a_nz
    have le_deriv := Polynomial.degree_le_of_dvd h deriv_nz
    have lt_self := le_deriv.trans_lt deriv_lt
    simp only [lt_self_iff_false] at lt_self
  mpr h := simp [h]
```

Reply:
Thanks, I just learned `mp` and `mpr` ;D